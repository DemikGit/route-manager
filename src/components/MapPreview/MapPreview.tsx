import * as React from 'react';
import { Component } from 'react';
import { GoogleMap, withGoogleMap, withScriptjs } from "react-google-maps"
import './MapPreview.css'


export interface IMapPreviewProps {
  children?: React.ReactNode,
  mountMap: React.RefObject<GoogleMap>,
  defaultCenter: {
    lat: number,
    lng: number,
  }
  defaultZoom: number,
}

class MapPreviewComponent extends Component<IMapPreviewProps, object> {
  public render() {

    const {
      mountMap,
      children,
      defaultCenter,
      defaultZoom,
    } = this.props;

    return (
      <GoogleMap
        defaultZoom={ defaultZoom }
        ref={ mountMap }
        mapTypeId={ google.maps.MapTypeId.HYBRID }
        defaultCenter={ defaultCenter }
      >
        { children && children }
      </GoogleMap>
    );
  }
}

export const MapPreview = withScriptjs(
  withGoogleMap (
    MapPreviewComponent
  )
);
