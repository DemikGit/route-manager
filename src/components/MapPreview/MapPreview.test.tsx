import { shallow, mount, render } from 'enzyme';
import 'jest-enzyme';
import * as React from 'react';
import { MapPreview } from './MapPreview';

it('renders without crashing', () => {
  const wrapper = mount(
    <MapPreview
      googleMapURL={ 'google.api' }
      defaultCenter={{
        lat: 37.666030368995685,
        lng: -116.02545749206541,
      }}
      defaultZoom={ 15 }
      loadingElement={ <div style={{ height: '100%' }} /> }
      containerElement={
        <div className="route-manager__map__wrapper"/>
      }
      mapElement={ <div style={{ height: '100%' }} /> }
      mountMap={ this.mapRef }
    />
  );

  expect(wrapper).toMatchSnapshot();
});
