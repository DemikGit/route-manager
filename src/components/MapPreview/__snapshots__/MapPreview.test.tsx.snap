// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<withScriptjs(withGoogleMap(MapPreviewComponent))
  containerElement={
    <div
      className="route-manager__map__wrapper"
    />
  }
  defaultCenter={
    Object {
      "lat": 37.666030368995685,
      "lng": -116.02545749206541,
    }
  }
  defaultZoom={15}
  googleMapURL="google.api"
  loadingElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
  mapElement={
    <div
      style={
        Object {
          "height": "100%",
        }
      }
    />
  }
>
  <div
    style={
      Object {
        "height": "100%",
      }
    }
  />
</withScriptjs(withGoogleMap(MapPreviewComponent))>
`;
